### PR TITLE
[REVIEW] 10 Minutes to cuDF and dask_cuDF

### PR DESCRIPTION
**Pull request by beckernick**
**Summary of Changes**
- This PR creates a 10 minute guide to both cuDF and dask_cuDF. The tutorial takes a paired approach, showing how to one action in both cuDF and dask_cuDF before moving to another action.

This can supersede the 10 Minute guide to cuDF if it's desirable.

### CODE REVIEW

**Comment by beckernick** [comment]
@randerzander this is ready for initial review. There are a couple of questions in the notebook as comments in the cells.

File: docs/source/10min_dask_cudf.ipynb [hunk 1]
```diff
@@ -0,0 +1,2561 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "10 Minutes to cuDF and dask_cuDF\n",
+    "=======================\n",
+    "\n",
+    "Modeled after 10 Minutes to Pandas, this is a short introduction to cuDF and dask_cuDF, geared mainly for new users."
```
Comment (randerzander) [inline docs/source/10min_dask_cudf.ipynb]: We could use some explanatory text here about:
1. What is cuDF
2. What is Dask
3. What is dask-cuDF

How do these interact, and what are they used for? For example, I would use dask-cudf when I want to use multiple GPUs, or when I have more data than can fit in memory on a single GPU, and I don't want to write all sorts of code to manage looping through files and processing iteratively

**Comment by beckernick** [comment]
Updated to reflect reviews. One cell is now blocked by a new issue rapidsai/dask-cudf#125.

@randerzander , I think this should be in index.rst instead of api.rst. It's not part of the API docs but the broader documentation, like the other notebooks. I'm going to supersede the existing `10min.ipynb`, but leave the file for now.

Relatedly, what are the necessary steps to propagate this to docs.rapids.ai?

File: docs/source/10min_dask_cudf.ipynb (hunk 1 above)
Comment (beckernick) [reply docs/source/10min_dask_cudf.ipynb]: Added a brief introduction to the libraries.

**Comment by randerzander** [comment]
> Relatedly, what are the necessary steps to propagate this to docs.rapids.ai?

When this PR merges, cudf docs are auto-built and published to docs.rapids.ai in the cuDF nightly API section.

File: docs/source/10min_dask_cudf.ipynb [hunk 2]
```diff
@@ -0,0 +1,2565 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "10 Minutes to cuDF and dask_cuDF\n",
```
Comment (kkraus14) [inline docs/source/10min_dask_cudf.ipynb]: Dask libraries in general use dashes between dask and the name, regardless of Python module name.
```suggestion
    "10 Minutes to cuDF and Dask-cuDF\n",
```

File: docs/source/10min_dask_cudf.ipynb [hunk 3]
```diff
@@ -0,0 +1,2565 @@
...
+    "\n",
+    "Modeled after 10 Minutes to Pandas, this is a short introduction to cuDF and dask_cuDF, geared mainly for new users.\n",
+    "\n",
+    "### What are these Libraries?\n",
+    "\n",
+    "[cuDF]() is a Python GPU DataFrame library (built on the Apache Arrow columnar memory format) for loading, joining, aggregating, filtering, and otherwise manipulating data.\n",
+    "\n",
+    "[Dask]() is a flexible library for parallel computing in Python that makes scaling out your workflow smooth and simple.\n",
+    "\n",
+    "[Dask cuDF]() is library that provides a partitioned, GPU-backed dataframe, using Dask.\n",
```
Comment (kkraus14) [inline docs/source/10min_dask_cudf.ipynb]:
```suggestion
    "[Dask-cuDF]() is library that provides a partitioned, GPU-backed dataframe, using Dask.\n",
```

File: docs/source/10min_dask_cudf.ipynb [hunk 4]
```diff
@@ -0,0 +1,2565 @@
...
+    "\n",
+    "\n",
+    "### When to use cuDF and Dask cuDF\n",
+    "\n",
+    "If your workflow is fast enough enough on a single GPU or your data comfortably fits in memory on a single GPU, you would want to use cuDF. If you want to distribute your workflow across multiple GPUs, have more data than you can fit in memory on a single GPU, or want to analyze data spread across many files at once, you would want to use Dask cuDF."
```
Comment (kkraus14) [inline docs/source/10min_dask_cudf.ipynb]:
```suggestion
    "If your workflow is fast enough on a single GPU or your data comfortably fits in memory on a single GPU, you would want to use cuDF. If you want to distribute your workflow across multiple GPUs, have more data than you can fit in memory on a single GPU, or want to analyze data spread across many files at once, you would want to use Dask-cuDF."
```

File: docs/source/10min_dask_cudf.ipynb [hunk 5]
```diff
@@ -0,0 +1,2565 @@
...
+   "metadata": {},
+   "source": [
+    "## Boolean Indexing"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Selecting rows in a `DataFrame` or `Series` by direct Boolean indexing. Boolean indexing on `Series` objects containing nulls is not currently implemented."
```
Comment (kkraus14) [inline docs/source/10min_dask_cudf.ipynb]: This should be fixed in the latest 0.8 nightlies, we recently merged handling the nullmask in boolean masking 😄 
```suggestion
    "Selecting rows in a `DataFrame` or `Series` by direct Boolean indexing."
```

File: docs/source/10min_dask_cudf.ipynb [hunk 6]
```diff
@@ -0,0 +1,2565 @@
...
+   "metadata": {},
+   "source": [
+    "## String Methods"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Like pandas, cuDF provides string processing methods in the `str` attribute of `Series`. Full documentation of string methods is a work in progress. Please see the cuDF and nvstrings API documentation for more information."
```
Comment (kkraus14) [inline docs/source/10min_dask_cudf.ipynb]: My 2c but I don't want users using nvstrings directly as we have plans to refactor it out. cuDF should be the entrypoint for working with string data.
```suggestion
    "Like pandas, cuDF provides string processing methods in the `str` attribute of `Series`. Full documentation of string methods is a work in progress. Please see the cuDF API documentation for more information."
```

File: docs/source/10min_dask_cudf.ipynb [hunk 7]
```diff
@@ -0,0 +1,2565 @@
...
+   "source": [
+    "ds = dask_cudf.from_cudf(s, npartitions=2)\n",
+    "print(ds.str.lower().compute())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Merge\n",
```
Comment (kkraus14) [inline docs/source/10min_dask_cudf.ipynb]: Missing content?

File: docs/source/10min_dask_cudf.ipynb [hunk 8]
```diff
@@ -0,0 +1,2565 @@
...
+   ],
+   "source": [
+    "print(dgdf.grade.cat.codes.compute())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Plotting\n",
```
Comment (kkraus14) [inline docs/source/10min_dask_cudf.ipynb]: Missing content?

File: docs/source/10min_dask_cudf.ipynb [hunk 9]
```diff
@@ -0,0 +1,2565 @@
...
+   ],
+   "source": [
+    "print(ddf.groupby('agg_col1').agg({'a':'max', 'b':'mean', 'c':'sum'}).compute())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Reshaping\n",
```
Comment (kkraus14) [inline docs/source/10min_dask_cudf.ipynb]: This header is awkward in the notebook.

File: docs/source/10min_dask_cudf.ipynb [hunk 10]
```diff
@@ -0,0 +1,2565 @@
...
+   ],
+   "source": [
+    "print(ddf.compute().head().to_pandas())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Numpy"
```
Comment (kkraus14) [inline docs/source/10min_dask_cudf.ipynb]: Should add `df.to_matrix()` in this section as well.

File: docs/source/10min_dask_cudf.ipynb [hunk 11]
```diff
@@ -0,0 +1,2565 @@
...
+    "Writing parquet files using PyArrow under the hood."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 76,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ddf.compute().to_parquet('example_files')  "
```
Comment (kkraus14) [inline docs/source/10min_dask_cudf.ipynb]: We can't use dask-cudf `to_parquet` directly?

File: docs/source/10min_dask_cudf.ipynb (hunk 7 above)
Comment (beckernick) [reply docs/source/10min_dask_cudf.ipynb]: This is a weird one. Merge is the top level category for concat/join/append, but doesn't appear like it in the notebook.

```
Merge
------------

## Concat
```
This looks weird here, but is important for formatting in the actual docs.

File: docs/source/10min_dask_cudf.ipynb (hunk 10 above)
Comment (beckernick) [reply docs/source/10min_dask_cudf.ipynb]: Is there a `to_matrix` method for dataframes to go to numpy? I may be missing it but I don't think there is one. Do you mean `to_gpu_matrix` to grab the devicearray?

File: docs/source/10min_dask_cudf.ipynb (hunk 11 above)
Comment (beckernick) [reply docs/source/10min_dask_cudf.ipynb]: Good call. We can. We just can't read it back in directly with dask-cudf yet

File: docs/source/10min_dask_cudf.ipynb (hunk 2 above)
Comment (beckernick) [reply docs/source/10min_dask_cudf.ipynb]: 👍

File: docs/source/10min_dask_cudf.ipynb (hunk 10 above)
Comment (beckernick) [reply docs/source/10min_dask_cudf.ipynb]: Addressed all comments except this one. Can include `as_gpu_matrix`, or something else. Defer to you @kkraus14

Comment (beckernick) [reply docs/source/10min_dask_cudf.ipynb]: Adding in `as_matrix` to convert to a standard numpy array.

**Comment by beckernick** [comment]
Could one of the @rapidsai/cudf-python-codeowners give this a review, too? Thanks!

File: docs/source/10min_dask_cudf.ipynb (hunk 10 above)
Comment (kkraus14) [reply docs/source/10min_dask_cudf.ipynb]: There is an `as_matrix()` method which is what I was referring to but typoed.

**Pull request merged**